nucypher/blockchain/eth/agents.py

```diff
@@ -435,23 +423,25 @@ def construct_batch_deposit_parameters(self, deposits: Dict[str, List[Tuple[int,
 
     @contract_api(TRANSACTION)
     def batch_deposit(self,
-                      stakers: list,
-                      number_of_substakes: list,
-                      amounts: list,
-                      lock_periods: list,
-                      sender_address: str,
+                      stakers: List[ChecksumAddress],
+                      number_of_substakes: List,
+                      amounts: List[NuNits],
+                      lock_periods: PeriodDelta,
+                      sender_address: ChecksumAddress,
                       dry_run: bool = False,
-                      gas_limit: int = None) -> Union[dict, int]:
-        min_gas_batch_deposit = 250_000
+                      gas_limit: Optional[Wei] = None
+                      ) -> Union[TxReceipt, Wei]:
+        
+        min_gas_batch_deposit: Wei = Wei(250_000)  # TODO: move elsewhere?
         if gas_limit and gas_limit < min_gas_batch_deposit:
             raise ValueError(f"{gas_limit} is not enough gas for any batch deposit")
 
-        contract_function = self.contract.functions.batchDeposit(stakers, number_of_substakes, amounts, lock_periods)
+        contract_function: ContractFunction = self.contract.functions.batchDeposit(stakers, number_of_substakes, amounts, lock_periods)
         if dry_run:
-            payload = {'from': sender_address}
+            payload: TxParams = {'from': sender_address}
             if gas_limit:
                 payload['gas'] = gas_limit
-            estimated_gas = contract_function.estimateGas(payload)  # If TX is not correct, or there's not enough gas, this will fail.
+            estimated_gas: Wei = Wei(contract_function.estimateGas(payload))  # If TX is not correct, or there's not enough gas, this will fail.
```

Comment:
gas measured in just int, I mean Wei confuses me at it's more related to gas price